### ISSUE TITLE
new chrome extension permissions requires identity.email

### ISSUE THREAD

**Issue by budevg** [ZachSaucier/Just-Read#339]
Hi,
Thanks for the awesome product. 

I noticed that recently chrome manifest was changed to require identity.email permission 
[here](https://github.com/ZachSaucier/Just-Read/blob/e994f4ccd02ff7ea8b17167e041ea5b8ac3a9362/manifest.json#L66)

Since providing email to extension without real need might be considered a security issue, 
is it possible to somehow avoid asking for such broad permissions. I see that this is used to send users uid to "just read" site [here](https://github.com/ZachSaucier/Just-Read/blob/e994f4ccd02ff7ea8b17167e041ea5b8ac3a9362/messager.js#L3)
and data uid collected [here](https://github.com/ZachSaucier/Just-Read/blob/e994f4ccd02ff7ea8b17167e041ea5b8ac3a9362/background.js#L222)

**Comment by ZachSaucier** [comment]
Hey budevg. 
TLDR: this requirement will be removed soon.

Long story: 
Because Chrome is deprecating their payment platform, I had to create my own. Because I was creating my own, I could make it in an-extension purchase so that I don't have to maintain two separate versions of Just Read and could make upgrading to Premium easier. 

The old Premium extension used a user's Chrome identity to verify that the user had Premium. The new system doesn't do that, but it still needs the old identity to find any shared pages that the user created if they had Premium. Once the Premium users have migrated to the new system this functionality can and will be removed. But I added the identity functionality to the free version for now to help aid in that transition (in case a user uninstalls the Premium version before they sign up for an account in the new system). 

Does that make sense? In a month or so I will remove the old Premium extension from the webstore and remove the identity requirements for the free version.

**Comment by ZachSaucier** [comment]
Given the fairly large amount of push back this has received here and in the reviews I've gone ahead and removed the permissions for this.

**Comment by hyphenminus** [comment]
> Given the fairly large amount of push back this has received here and in the reviews I've gone ahead and removed the permissions for this.

Thanks for fixing this Zach. I see there had been some comments in the Chrome Web Store https://chrome.google.com/webstore/detail/just-read/dgmanlpmmkibanfdgjocnabmcaclkmod but the latest version (5.0.4) has just appeared in my browser (on clicking Update from the Chrome Extensions page), and this version indeed fixes the permissions request problem. I'm now seeing some slight changes in the appearance of some articles, which might be caused by my custom CSS or perhaps by minor bugs (in which case I expect you will also notice them in due course).

Thank you for your continued work on this and your quick response, making the Web readable again!

**Comment by ZachSaucier** [comment]
Thanks for the support. Some of the reviews today have been rough. 

I'm guessing any CSS issues that have appeared are in custom themes you have? The selectors changed slightly so existing themes may require some adjustments.

**Comment by hyphenminus** [comment]
> Thanks for the support. Some of the reviews today have been rough.
> 
> I'm guessing any CSS issues that have appeared are in custom themes you have? The selectors changed slightly so existing themes may require some adjustments.

Thanks for the swift further response. My layout is heavily customised, so I am assuming that the quirks which I am currently noticing are for me to fix unless you are already noticing similar things yourself :)

Now that the permissions have been fixed so the extension itself is again working (version 5.0.4), I have only just realised that the Options page still requires an email sign-up. I am not clear from your recent github comments whether this barrier is also something you intend to revert, or whether it is intentional. It does seem odd that, to access the Options even on the free version, an email address is required (via the http://justread.link/#getJustRead link), especially as your [privacy statement](https://github.com/ZachSaucier/Just-Read#privacy-statement) implies that the email address is stored but not actually used. Obviously it is good that you are promising not to use the email address, and of course you are entitled to require people to register for the premium version (or to prevent customisation for non-registered users, though that would be a shock!), but the current Options restriction is puzzling given the data protection principle of not requiring or storing unnecessary personal information.

**Comment by ZachSaucier** [comment]
It was intended as a way to get more email addresses so I can market premium better and such. But at this point I'll probably remove the restriction. Who knew people would be so upset about providing an email address.

**Comment by hyphenminus** [comment]
> It was intended as a way to get more email addresses so I can market premium better and such. But at this point I'll probably remove the restriction. Who knew people would be so upset about providing an email address.

Thanks. Though you have always been very open and active in responding to feedback, any extension requires a lot of trust from users who know nothing about code audit, and there have been other Chrome extensions which have infamously been written or later taken over to behave unscrupulously, or where legitimate email lists have been hacked. So I hope you will not take it personally that some people are cautious or even alarmed by unusual messages or unexplained requests for data which could be misused by others even though that was not your intent. I guess that even use for upgrade-marketing would require some kind of informed consent.

**Comment by ZachSaucier** [comment]
I've removed the sign up need in 3.0.5.

**Comment by hyphenminus** [comment]
Great, 3.0.5 looks good to me, with the Options now freely accessible and saveable again.

I added `.simple-article-container {padding-top: 0; display: block}` to my existing custom CSS and that fixed the visual quirks I had noticed in 3.0.4 as a result of the interaction of your new styles with my own non-standard layout. Thanks again.

**Comment by budevg** [comment]
Thanks for fixing this so quickly and thanks again for the awesome job you are doing with just read extension.
